Foundation/FileManager+POSIX.swift

```diff
@@ -0,0 +1,778 @@
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+#if !os(Windows)
+
+
+#if os(Android) // struct stat.st_mode is UInt32
+internal func &(left: UInt32, right: mode_t) -> mode_t {
+    return mode_t(left) & right
+}
+#endif
+
+import CoreFoundation
+
+extension FileManager {
+    internal func _mountedVolumeURLs(includingResourceValuesForKeys propertyKeys: [URLResourceKey]?, options: VolumeEnumerationOptions = []) -> [URL]? {
+        var urls: [URL] = []
+
+#if os(Linux)
+        guard let procMounts = try? String(contentsOfFile: "/proc/mounts", encoding: .utf8) else {
+            return nil
+        }
+        urls = []
+        for line in procMounts.components(separatedBy: "\n") {
+            let mountPoint = line.components(separatedBy: " ")
+            if mountPoint.count > 2 {
+                urls.append(URL(fileURLWithPath: mountPoint[1], isDirectory: true))
+            }
+        }
+#elseif canImport(Darwin)
+
+        func mountPoints(_ statBufs: UnsafePointer<statfs>, _ fsCount: Int) -> [URL] {
+            var urls: [URL] = []
+
+            for fsIndex in 0..<fsCount {
+                var fs = statBufs.advanced(by: fsIndex).pointee
+
+                if options.contains(.skipHiddenVolumes) && fs.f_flags & UInt32(MNT_DONTBROWSE) != 0 {
+                    continue
+                }
+
+                let mountPoint = withUnsafePointer(to: &fs.f_mntonname.0) { (ptr: UnsafePointer<Int8>) -> String in
+                    return string(withFileSystemRepresentation: ptr, length: strlen(ptr))
+                }
+                urls.append(URL(fileURLWithPath: mountPoint, isDirectory: true))
+            }
+            return urls
+        }
+
+        if #available(OSX 10.13, *) {
+            var statBufPtr: UnsafeMutablePointer<statfs>?
+            let fsCount = getmntinfo_r_np(&statBufPtr, MNT_WAIT)
+            guard let statBuf = statBufPtr, fsCount > 0 else {
+                return nil
+            }
+            urls = mountPoints(statBuf, Int(fsCount))
+            free(statBufPtr)
+        } else {
+            var fsCount = getfsstat(nil, 0, MNT_WAIT)
+            guard fsCount > 0 else {
+                return nil
+            }
+            let statBuf = UnsafeMutablePointer<statfs>.allocate(capacity: Int(fsCount))
+            defer { statBuf.deallocate() }
+            fsCount = getfsstat(statBuf, fsCount * Int32(MemoryLayout<statfs>.stride), MNT_WAIT)
+            guard fsCount > 0 else {
+                return nil
+            }
+            urls = mountPoints(statBuf, Int(fsCount))
+        }
+#else
+#error("Requires a platform-specific implementation")
+#endif
+        return urls
+    }
+ internal func _urls(for directory: SearchPathDirectory, in domainMask: SearchPathDomainMask) -> [URL] {
```

Comment:
I think the per-OS helper functions should be `private` instead of `internal` where possible`. These are probably a holdover from when they were first written but may as well fix them now.

Reply:
I believe most of these need to be internal instead of private since they are used in a file they aren't defined in.

Reply:
Ah yeah, good point.